pkg/client/unstructured_client.go

```diff
@@ -111,7 +111,7 @@ func (uc *unstructuredClient) Delete(ctx context.Context, obj Object, opts ...De
 		Name(o.GetName()).
 		Body(deleteOpts.AsDeleteOptions()).
 		Do(ctx).
-		Error()
+		Into(obj)
```

Comment:
I ran this through the debugger and this object looks like this afterwards if I'm not mistaken:

![image](https://github.com/user-attachments/assets/77a14af8-b858-45c5-908c-48b4225e7472)

I.e. the delete endpoint does not return a resource, it returns a status

Reply:
I think we also need a test case that when we delete via Unstructured without a finalizer the object keeps it's data and does not become a `Status` object

Reply:
Should we maybe just check if the returned body is a status and only if its not update the original object?

Reply:
As far as I can tell there's no way to tell apart you got the object back or a metav1.Status unless you actually read the request body.

Reply:
Which I think might be okay, the main overhead will be that we have to deserialize it twice if it is not a status

Reply:
I don't remember the details to be honest. Does anyone know if we would be able to get the deletionTimestamp in all cases? (or only if there are finalizers or not). The latter would be pretty surprising behavior

Reply:
So it is possible to get a status back, even if the object is still in the system?

I wonder if it is still worth to check if the result is not a status and update `obj` if so

Reply:
I think it's a bad UX if after calling client.Delete() you only sometimes see the deletionTimestamp set :/

Reply:
Since Delete() already doesn't do what we expect it to do, what if we changed its return signature to `(deletedFromStorage bool, err error)`?

Reply:
> I mean what can we realistically put in there if the object is gone from storage? I guess we could put a non-nil fake deletiontimestamp in there but not sure if that is a good idea.

Yup agree. I mostly just meant that if deletionTimestamp is only sometimes set (aka in some cases) this would be tricky to rely on